tests/slow/src/test/scala/tests/bazel/BazelLspSuite.scala

```diff
@@ -0,0 +1,167 @@
+import tests.BazelBuildLayout
+import tests.BazelServerInitializer
+
+class BazelLspSuite
+    extends BaseImportSuite("bazel-import", BazelServerInitializer) {
+  val buildTool: BazelBuildTool = BazelBuildTool(() => userConfig, workspace)
+
+  val bazelVersion = "6.4.0"
+
+  def bazelBspConfig: AbsolutePath = workspace.resolve(".bsp/bazelbsp.json")
+
+  override def currentDigest(
+      workspace: AbsolutePath
+  ): Option[String] = BazelDigest.current(workspace)
+
+  test("basic") {
+    cleanWorkspace()
+    for {
+      _ <- initialize(
+        BazelBuildLayout(workspaceLayout, V.scala213, bazelVersion)
+      )
+      _ = assertNoDiff(
+        client.workspaceMessageRequests,
+        List(
+          importBuildMessage,
+          // create .bazelbsp progress message
+          BazelBuildTool.mainClass,
+          bazelNavigationMessage,
+        ).mkString("\n"),
+      )
+      _ = assert(bazelBspConfig.exists)
+      _ = client.messageRequests.clear() // restart
+      _ = assertStatus(_.isInstalled)
+      _ <- server.didChange("WORKSPACE")(_ + "\n# comment")
+      _ <- server.didSave("WORKSPACE")(identity)
+      // Comment changes do not trigger "re-import project" request
+      _ = assertNoDiff(client.workspaceMessageRequests, "")
+      _ <- server.didChange(s"BUILD") { text =>
+        text.replace("main", "main1")
+      }
+      _ = assertNoDiff(client.workspaceMessageRequests, "")
+      _ = client.importBuildChanges = ImportBuildChanges.yes
+      _ <- server.didSave(s"BUILD")(identity)
+    } yield {
+      assertNoDiff(
+        client.workspaceMessageRequests,
+        List(
+          importBuildChangesMessage
+        ).mkString("\n"),
+      )
+      server.assertBuildServerConnection()
+    }
+  }
+
+  test("generate-bsp-config") {
+    cleanWorkspace()
+    writeLayout(
+      BazelBuildLayout(workspaceLayout, V.scala213, bazelVersion)
+    )
+    for {
+      _ <- server.initialize()
+      _ <- server.initialized()
+      _ = assertNoDiff(client.workspaceMessageRequests, importBuildMessage)
+      _ = client.messageRequests.clear()
+      // We dismissed the import request, so bsp should not be configured
+      _ = assert(!bazelBspConfig.exists)
+      _ <- server.didChange(s"BUILD") { text =>
+        text.replace("runner", "runner1")
+      }
+      _ <- server.didSave(s"BUILD")(identity)
+      _ = assertNoDiff(client.workspaceMessageRequests, "")
+      _ = server.server.buildServerPromise = Promise()
+      _ <- server.executeCommand(ServerCommands.GenerateBspConfig)
+      // We need to wait a bit just to ensure the connection is made
+      _ <- server.server.buildServerPromise.future
+    } yield {
+      assertNoDiff(
+        client.workspaceMessageRequests,
+        List(
+          "bazelbsp bspConfig",
+          bazelNavigationMessage,
+        ).mkString("\n"),
+      )
+      assert(bazelBspConfig.exists)
+      server.assertBuildServerConnection()
+    }
+  }
+
+  test("import-build") {
```

Comment:
Could we add a check that the code actually compiles? I am unable to get it working locally actually, seems the dependencies are not seen. Also it's not possible to get information about a specific build target `Build target :main1 not found`

Reply:
What do you think about this? We can assert that there are not diagnsotics.